Terminal.Gui/Text/TextFormatter.cs

```diff
@@ -304,7 +304,7 @@ public bool WordWrap
             {
                 if (isVertical)
                 {
-                    int runesWidth = GetWidestLineLength (linesFormatted, line, TabWidth);
+                    int runesWidth = GetColumnsRequiredForVerticalText (linesFormatted, tabWidth: TabWidth);
```

Comment:
@tig the `GetWidestLineLength` method here in the `Draw` method call the one that has four parameters. All theses parameters are necessary to be passed depending on the vertical alignment used. Please see my PR #3429 where the centered and right alignments on the horizontal direction, and where the middle and bottom vertical alignments on the vertical direction, are working as expected.

```cs
    public static int GetWidestLineLength (
        List<string> lines,
        int startIndex = -1,
        int length = -1,
        int tabWidth = 0
```

Reply:
You also have to consider the word wrap behavior which doesn't exist yet in the `Text Alignment and Direction` scenario and in my PR was implemented.

See that by using the horizontal right alignment on a vertical direction text bug still persist in this PR.

![image](https://github.com/BDisp/Terminal.Gui/assets/13117724/6dba8643-c9ec-4861-a804-c36517b21afc)

Reply:
Anyway, I don't know what the cursor has to do with the text format. It would make more sense for changes to the text format to be submitted in my respective PR and not in this one.